Source/WebGPU/WebGPU/RenderPipeline.h

```diff
@@ -36,14 +37,20 @@ namespace WebGPU {
 
 class BindGroupLayout;
 class Device;
+class PipelineLayout;
 
 // https://gpuweb.github.io/gpuweb/#gpurenderpipeline
 class RenderPipeline : public WGPURenderPipelineImpl, public RefCounted<RenderPipeline> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    static Ref<RenderPipeline> create(id<MTLRenderPipelineState> renderPipelineState, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, uint32_t vertexShaderInputBufferCount, Device& device)
+    static Ref<RenderPipeline> create(id<MTLRenderPipelineState> renderPipelineState, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, MTLRenderPipelineReflection *reflection, uint32_t vertexShaderInputBufferCount, Device& device)
```

Comment:
I’m reading your previous comment to mean that this is throwaway code that will be replaced soon

Reply:
Also, a bind group layout contains more information than metal reflection, so the sooner we can do this using WebGPU types, the better

Reply:
Yes, it would be great to get this information directly from the WGSL compiler